src/rules/awaitPromiseRule.ts

```diff
@@ -42,6 +42,7 @@ export class Rule extends Lint.Rules.TypedRule {
     /* tslint:enable:object-literal-sort-keys */
 
     public static FAILURE_STRING = "'await' of non-Promise.";
+    public static FAILURE_FOR_AWAIT_OF = "'for-await-of' of non-AsyncIterable.";
```

Comment:
I realized both of these error messages are a bit too terse -- how about 

- "Invalid 'await' of a non-Promise value"
- "Invalid 'for-await-of' of a non-AsyncIterable value"